util/collate/collate.go

```diff
@@ -139,6 +157,7 @@ func (bpc *binPaddingCollator) Key(str string, opt CollatorOption) []byte {
 func init() {
 	collatorMap = make(map[string]Collator)
 	collatorIDMap = make(map[int]Collator)
+	id2NameMap = make(map[int32]string)
```

Comment:
We have the same defined in `parser` already:
https://github.com/pingcap/parser/blob/517beb2e39c20e78cca59b63351aacb5c1ef5845/mysql/charset.go#L126

Please use this one instead.